### ISSUE TITLE
[404] Error getting Key-File when string has &

### ISSUE THREAD

**Issue by finelikewine** [selsta/hlsdl#81]
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-TARGETDURATION:5
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-KEY:METHOD=AES-128,URI="https://websiteurl.com?id=DF4s$t%FvsdEdc&fadk=couk"

Error: Getting key-file [https://websiteurl.com?id=DF4s$t%FvsdEdc&fadk=couk] failed http_code[404]

Not sure if it's a website issue, IP issue, or if it has to do with the & which is preventing HLSDL from getting the key.

**Comment by Pfuenzle** [comment]
nope, same error for me

**Comment by glgoose** [comment]
> nope, same error for me.
> Havent look into the code yet, but I hope I will be able to fix it, or at least make it possible to supply a keyfile via terminal

Yes, it would be great to have the ability to supply a keyfile via terminal

**Comment by xavave** [comment]
hi,
I've already made a few fixes and upgrades on my fork https://github.com/xavave/hlsdl-vs2019, but not this one
1- I would like now to be able to reproduce the issue with '&' key file, would you have a key file within a http(s) url example please?  (I would need the url)
2- for supplying a keyfile via terminal, what would you suggest ? this provided keyfile would override the original keyfile?
